llrt_core/src/modules/crypto/mod.rs

```diff
@@ -134,7 +137,16 @@ fn random_fill_sync<'js>(
 }
 
 fn get_random_values<'js>(ctx: Ctx<'js>, obj: Object<'js>) -> Result<Object<'js>> {
-    if let Some((array_buffer, source_length, source_offset)) = obj_to_array_buffer(&obj)? {
+    if let Some(object_bytes) = ObjectBytes::from_array_buffer(&obj)? {
+        if matches!(
+            object_bytes,
+            ObjectBytes::F64Array(_) | ObjectBytes::F32Array(_)
+        ) {
+            return Err(Exception::throw_message(&ctx, "Unsupported TypedArray"));
+        }
+
+        let (array_buffer, source_length, source_offset) =
+            object_bytes.get_array_buffer()?.unwrap();
```

Comment:
Not a fan of unwrap even if this is safe right now. Just takes a refactor to making it crash. At minimum lets use an expect or better turn that into a result.